psutil/arch/windows/socks.c

```diff
@@ -56,6 +68,7 @@ static DWORD __GetExtendedTcpTable(TYPE_GetExtendedTcpTable call,
         PyErr_SetString(PyExc_RuntimeError, "GetExtendedTcpTable failed");
```

Comment:
```suggestion
        if (*data) {
            free(*data);
            *data = NULL;
        }
        PyErr_SetString(PyExc_RuntimeError, "GetExtendedTcpTable failed");
```
I should include it to avoid memory leak but I want to know if you like the idea behind the two changes. I can split them in two different Pull requests if you prefer.